core/src/main/java/apoc/nodes/Nodes.java

```diff
@@ -69,15 +69,19 @@
     public Pools pools;
 
     @Procedure(mode = Mode.WRITE)
-    @Description("apoc.nodes.link([nodes],'REL_TYPE') - creates a linked list of nodes from first to last")
-    public void link(@Name("nodes") List<Node> nodes, @Name("type") String type) {
+    @Description("apoc.nodes.link([nodes],'REL_TYPE', conf) - creates a linked list of nodes from first to last")
+    public void link(@Name("nodes") List<Node> nodes, @Name("type") String type, @Name(value = "config",defaultValue = "{}") Map<String, Object> config) {
+        RefactorConfig conf = new RefactorConfig(config);
         Iterator<Node> it = nodes.iterator();
         if (it.hasNext()) {
             RelationshipType relType = RelationshipType.withName(type);
             Node node = it.next();
             while (it.hasNext()) {
                 Node next = it.next();
-                node.createRelationshipTo(next, relType);
+                final boolean checkAndNotExists = conf.isAvoidDuplicates() && connected(node, next, type);
```

Comment:
ragioniamo un attimo perché abbiamo sempre il solito problema di nomenclatura di variabili e metodi.
Ti devi sempre fare questa domanda: Qual'è la finalità della variabile/metodo?
in questo caso dobbiamo controllare se inserire una relazione:
```suggestion
                final boolean createRelationship = !conf.isAvoidDuplicates() || (conf.isAvoidDuplicates() && !connected(node, next, type));
                if (createRelationship) {
```